### DISCUSSION TITLE
Vehicles (mod ?)

### DISCUSSION THREAD

**Discussion by FabioOG** [redeclipse/base#1141] [Ideas and Feature Requests]
I know it will be quite complicated, but i thought that red eclipse could improve itself by making vehicles, even if it's only in a modded version. The way I imagined it was a pickable weapon (as grenades) with a skin of the size of the vehicle. With the new version we could change the weight of the weapon to a high value, then it could not jump or use jumppads. Is it something possible ?

**Comment by SniperGoth** [comment]
This was suggested for various times through the years, and personally i really don't think vehicles should be on the game.
Doesn't really fit the gameplay.

**Comment by cloudziq** [comment]
I even started to make a motorcycle for RE ( in hope that someday we can play on bigger maps) but project was scrapped due to many reasons..

**Comment by JEmanuel1806** [comment]
it would be an awesome mod idea nevertheless

**Comment by no-lex** [comment]
Let's take a look at RE's physics:

- Player runs at about 60kph (37mph) _before_ impulse moves are taken into account;
- Players can change 100% of their momentum instantaneously via impulse moves;
- Players mostly are able to avoid dying due to their agility, not their ability to tank damage;
- Tesseract is an indoor engine and has performance issues as map sizes increase;
- Maps are designed around being able to redirect momentum at will.

Vehicles generally:
- Have greater momentum than individual players, because they are more massive; take time to start, stop, and turn around;
- Are relatively easy to hit but are able to tank significant amounts of damage before dying;
- Able to deal large amounts of damage;
- Are incapable of climbing ladders/stairs/jumpers.

Let's examine the issues with these traits as they pertain to RE:
 - RE has never been designed around momentum as an gameplay characteristic; maps have not been designed around players with significant momentum, leading to choppy, clumsy vehicle gameplay.
 - RE teamplay does not have significant focus on a tug-of-war style battle where teams fight back-and-forth between established sides; instead, gameplay is focused on quickly streaking into opponent's bases to capture objectives. Vehicles in this style of gameplay are therefore avoided completely due to layouts being nonlinear, or constrained to largely static defense of a base, which is not a worthwhile reason for their inclusion.
 - Players in RE are already plenty capable of dealing large amounts of damage, and their aim is the only thing preventing instagib matches in every gamemode. Vehicles would therefore require absurd ROF or splash damage bonuses in order to be notably more effective, which is difficult to impossible to balance.
 - Almost all RE maps rely heavily on jumppads, stairs, and ladders to create verticality within the level. Vehicles are unsuitable for these features due to their size and lack of vertical mobility. Flying vehicles are nightmare to deal with both in terms of balance and in terms of map clipping.

Additionally, there are technical problems with implementing vehicles:

- Vehicles would require large amounts of animation, ranging from wheels (if present) to particle effects, player entrance/exit and weapon firing. There is simply not adequate resources available to service such a difficult and complex task to the extent required for release-worthy content.

 - Vehicles would require their own actor type, complete with its own hitbox and mobility concerns that have not been addressed with any other actor type. Vehicles, in order to be properly realistic, have to faithfully restrict their movement as a normal wheeled or tracked vehicle would. This would require fixing the direction of movement to the current direction of the steering wheels (which would require its own animations).

 - Vehicles would have to be well-made, high quality assets (at least on par with salatiel's spaceships) and therefore may be a performance concern. 

 - Glass on vehicles is problematic due to Tesseract's renderer not being capable of more than one alpha level.

Remember that RE is an open-source game and you are free to do whatever you want with it on your own time; if you wish to undertake the task on your own then there is nothing stopping you from doing so. 

This idea has come up many times, and while it is a neat concept when properly executed in the proper type of game, RE is neither the proper platform nor has the resources to properly implement vehicles.

**Comment by FabioOG** [comment]
Yes, i know it's true that it's hard to build them and to adapt them to the gameplay, but there will be present in some new maps, appropriated for vehicles. The idea i had of making them a weapon would be that you can still be a player and have the vehicle in your inventory. I think we can try to limit the number of animation and try no to make alpha to reduce the lag. Maybe i'll try to understand how the code for actors of for weapon models work, but i don't think i will succeed to do the mod
But i really want to see what you have done Dziq
I also thought there could be a vehicle mutator, for activating or not the vehicles in map where there are some for players who don't like them.

**Comment by cloudziq** [comment]
Nothing worth showing - also please do not make multiple posts, there is a EDIT function (three dots), Thank you and have a nice day

**Comment by FabioOG** [comment]
I think as i thought understand in the actor code (i didn't understand very much) that the actors are programmed to aim players, so a vehicle shouldn't be an independent type of entity ?
Can someone teach me how works the libraries of red eclipse, i don't know anything about them.
